packages/next-swc/crates/napi/src/lib.rs

```diff
@@ -105,15 +111,17 @@ fn init() {
         if cfg!(debug_assertions) || env::var("SWC_DEBUG") == Ok("1".to_string()) {
             eprintln!("{}", info);
         } else {
-            let size = std::fs::metadata(LOG_FILE_PATH).map(|m| m.len());
+            let size = std::fs::metadata(PANIC_LOG.as_path()).map(|m| m.len());
             if let Ok(size) = size {
                 if size > 512 * 1024 {
                     // Truncate the earliest error from log file if it's larger than 512KB
                     let new_lines = {
                         let log_read = OpenOptions::new()
                             .read(true)
-                            .open(LOG_FILE_PATH)
-                            .unwrap_or_else(|_| panic!("Failed to open {}", LOG_FILE_PATH));
+                            .open(PANIC_LOG.as_path())
+                            .unwrap_or_else(|_| {
+                                panic!("Failed to open {}", PANIC_LOG.to_string_lossy())
+                            });
```

Comment:
Out of curiosity... What happens if you panic inside of the panic hook?

Reply:
iirc (I ran into this while debugging) the process just exits with a generic panic message and it's not handled by the hook.